Sources/ClientRuntime/Networking/Streaming/BufferedStream.swift

```diff
@@ -107,4 +131,21 @@ public class BufferedStream: Stream {
             isClosed = true
         }
     }
+
+    // If there is a client waiting to read data, and there is unread data remaining in the buffer,
+    // read the data and pass it to the reader via the continuation.
+    // Continue until there are no clients or no data.
+    private func _flushToNextClient() {
+        while !clients.isEmpty {
+            let (continuation, count) = clients[0]  // Don't remove the client from the array yet
+            do {
+                guard let data = try _read(upToCount: count), !data.isEmpty else { return }
```

Comment:
If `data` is `nil`, we should still resume the continuation as `nil` marks the end of the stream.